test/integration/controller/jobset_controller_test.go

```diff
@@ -643,6 +644,34 @@ func matchJobsSuspendState(js *jobset.JobSet, suspend bool) (bool, error) {
 	return true, nil
 }
 
+func matchJobsNodeSelectors(js *jobset.JobSet, nodeSelectors map[string]map[string]string) (bool, error) {
+	var jobList batchv1.JobList
+	if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
+		return false, err
+	}
+	// Count number of changed jobs
+	jobsChanged := 0
+	for _, job := range jobList.Items {
+		nodeSelectorKey, ok := job.Labels[jobset.ReplicatedJobNameKey]
+		if !ok {
+			return false, fmt.Errorf(fmt.Sprintf("%s job missing ReplicatedJobName label", job.Name))
+		}
+		if apiequality.Semantic.DeepEqual(job.Spec.Template.Spec.NodeSelector, nodeSelectors[nodeSelectorKey]) {
```

Comment:
Should this logic be inverted (i.e., if node selectors are NOT equal, then increment `jobsChanged`)?

Reply:
this is tracking the jobs that got the expected change